saic-java-mqtt-gateway/src/main/java/net/heberling/ismart/mqtt/VehicleState.java

```diff
@@ -653,11 +654,28 @@ public void setRefreshPeriodAfterShutdown(long refreshPeriodAfterShutdown) {
     this.refreshPeriodAfterShutdown = refreshPeriodAfterShutdown;
   }
 
+  public void setRemoteTemperature(int remoteTemperature) {
+    if ( remoteTemperature == 0 || this.remoteTemperature != remoteTemperature) {
+
+      MqttMessage mqttMessage =
+              new MqttMessage(
+                      String.valueOf(remoteTemperature).getBytes(StandardCharsets.UTF_8));
+      try {
+        mqttMessage.setRetained(true);
+        this.client.publish(this.mqttVINPrefix + "/" + CLIMATE_REMOTE_TEMPERATURE, mqttMessage);
+      } catch (MqttException e) {
+        throw new MqttGatewayException("Error publishing message: " + mqttMessage, e);
+      }
+    }
+    this.remoteTemperature = remoteTemperature;
+  }
+
   public boolean isComplete() {
     return refreshPeriodActive != null
         && refreshPeriodInactive != null
         && refreshPeriodAfterShutdown != null
-        && refreshMode != null;
+        && refreshMode != null
+        && remoteTemperature != 0;
```

Comment:
!= null, See above

Reply:
if we change to Integer of course. ;)

Reply:
I am also thinking of refactoring this part, the long switch cases plus this long if smells like some kind of strategy pattern... :) But will first try to get this to work properly at least on typical car configurations, and then I will look into a better pattern...